packages/nimble-components/src/select/tests/select.spec.ts

```diff
@@ -1399,6 +1426,173 @@ describe('Select', () => {
         });
     });
 
+    describe('groups', () => {
+        let element: Select;
+        let connect: () => Promise<void>;
+        let disconnect: () => Promise<void>;
+        let pageObject: SelectPageObject;
+        const totalSlottedElements = 9;
+
+        beforeEach(async () => {
+            ({ element, connect, disconnect } = await setupWithGroups());
+            element.style.width = '200px';
+            await connect();
+            pageObject = new SelectPageObject(element);
+        });
+
+        afterEach(async () => {
+            await disconnect();
+        });
+
+        it('can select an option within a group', async () => {
+            await clickAndWaitForOpen(element);
+            pageObject.clickOption(1);
+            expect(element.value).toBe('two');
+            expect(element.selectedIndex).toBe(1);
+        });
+
+        it('clicking a group does not close the dropdown', async () => {
+            await clickAndWaitForOpen(element);
+            const group = element.querySelector('[role="group"]');
```

Comment:
Could you take a pass at moving group-related helper functions in the select page object? They could be useful to clients too. Anything does a `querySelector...('[role="group"]')` or interacts with the result of that would be a good candidate to move.

Reply:
I've added a couple of public APIs to the exported page object, but I've also create an internal page object that encapsulates all of the `querySelector` calls related to groups.